var/spack/repos/builtin/packages/py-tatsu/package.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyTatsu(PythonPackage):
+    """竜 TatSu (the successor to Grako) is a tool that takes grammars in
+    a variation of EBNF as input, and outputs memoizing (Packrat) PEG
+    parsers in Python."""
+
+    homepage = "https://github.com/neogeny/tatsu"
+    url      = "https://pypi.io/packages/source/T/TatSu/TatSu-4.4.0.zip"
+
+    version('4.4.0', sha256='80713413473a009f2081148d0f494884cabaf9d6866b71f2a68a92b6442f343d')
+
+    depends_on('python@3.6:', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
+    # part of the standard lib in python@3.7.0
+    depends_on('py-dataclasses@0.6:', type=('build', 'run'), when='^python@3.6.0:3.6.99')
+    depends_on('py-mypy@0.641:', type='test')
+    # optional dependency, otherwise falls back to standard implementation
+    depends_on('py-regex@2018.8:', type=('build', 'run'))
```

Comment:
I would add a `+future_regex` variant for this and make it optional. Also, drop the version requirement.

Reply:
Done, but I'm not sure whether this is a good policy, I think in general we should prefer keeping build receipts simple over small. `spack` is good at building things, and it's not as if an additional python library is a significant time consumer.

Reply:
Depends on how many packages are optional and which they are. Some packages like numpy/pandas take quite a while to build. For all packages (not just Python) I like to make all optional dependencies truly optional.